### PR TITLE
terminal,service: print WaitReason, WaitSince for goroutines

### LINKED ISSUE #637: Add wait reason to output of parked goroutines
For any sleeping goroutine we should be able to get the wait reason as to why they are parked. We should display this information in the `goroutines` output.

### PR DESCRIPTION
**Pull request by aarzilli**
Fixes #637

### CODE REVIEW

File: cmd/dlv/dlv_test.go [hunk 1]
```diff
@@ -322,9 +322,7 @@ func TestRedirect(t *testing.T) {
 
 	// and detach from and kill the headless instance
 	client := rpc2.NewClient(listenAddr)
-	if err := client.Detach(true); err != nil {
```
Comment (derekparker) [inline cmd/dlv/dlv_test.go:325]: I'm confused why this is included in this patch.

File: pkg/terminal/command.go [hunk 2]
```diff
@@ -917,11 +918,58 @@ func (t *Term) formatGoroutine(g *api.Goroutine, fgl formatGoroutineLoc) string
 		locname = "Start"
 		loc = g.StartLoc
 	}
-	thread := ""
+
+	buf := new(bytes.Buffer)
```
Comment (derekparker) [inline pkg/terminal/command.go:922]: We should use `strings.Builder` here. It's slightly more efficient in the `String` method, and shows intent a little better.

https://golang.org/pkg/strings/#Builder

File: pkg/terminal/command.go [hunk 3]
```diff
@@ -917,11 +918,58 @@ func (t *Term) formatGoroutine(g *api.Goroutine, fgl formatGoroutineLoc) string
...
+		if g.WaitSince > 0 {
+			fmt.Fprintf(buf, " %s", time.Since(time.Unix(0, g.WaitSince)).String())
+		}
+		fmt.Fprintf(buf, "]")
+	}
+
+	return buf.String()
+}
+
+var waitReasonStrings = [...]string{
```
Comment (derekparker) [inline pkg/terminal/command.go:945]: Is there ever a case we'd want this string outside of just the terminal? For example, returning wait reason as a string via API calls?

**Review by derekparker — CHANGES_REQUESTED** [review]

File: pkg/terminal/command.go (hunk 3 above)
Comment (dlsniper) [reply pkg/terminal/command.go:950]: As a potential consumer of the API (and the one that added support initially for this), I second @aarzilli's message.

Comment (derekparker) [reply pkg/terminal/command.go:950]: Ack, makes sense.

**Review by derekparker — APPROVED** [review]

**Pull request merged**